### PR TITLE
Improve IME interface

### PR DESCRIPTION
**Pull request by abarth**
Instead of providing a stream of mutations, we now provide a the client with
the complete state of the editing control to prevent the two from getting out
of sync.

As a side-effect, we can now move the cursor around when editing a text field.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde

File: sky/services/editing/editing.mojom [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  TextAffinity affinity = TextAffinity.downstream;
+  bool is_directional;
+};
+
+enum SubmitAction {
+  DONE,
+};
+
+enum KeyboardType {
+  TEXT,
```
Comment (chinmaygarde) [inline sky/services/editing/editing.mojom]: The casing of the names of the enum values is inconsistent (with `TextAffinity`)

Comment (abarth) [inline sky/services/editing/editing.mojom]: True.  I think mojom is supposed to be ALL_CAPS.

**Comment by chinmaygarde** [comment]
lgtm

**Comment by krisgiesing** [comment]
Why the deletion of the Linux keyboard code?

**Comment by abarth** [comment]
> Why the deletion of the Linux keyboard code?

We'll need that code eventually, but when we switched to Mozart the code became unreachable.  We'll need to integrate the code into Mozart so that all Mozart views can get a keyboard interface.

**Comment by abarth** [comment]
Also, I just deleted our duplicate copy of the code.  The code still exists in mojo.git, which is the proper place for it:

https://github.com/domokit/mojo/tree/master/services/keyboard/linux

**Pull request merged**